tests/compute/credentials.go

```diff
@@ -53,82 +53,91 @@ var _ = SIGDescribe("Guest Access Credentials", func() {
 		guestAgentConnetTimeout = 2 * time.Minute
 		denyListTimeout         = 2 * time.Minute
 		fedoraPassword          = "fedora"
+		pubKeySecretID          = "my-pub-key"
+		userPassSecretID        = "my-user-pass"
 	)
 
-	Context("with qemu guest agent", func() {
-		withQuestAgentPropagationMethod := v1.SSHPublicKeyAccessCredentialPropagationMethod{
-			QemuGuestAgent: &v1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
-				Users: []string{"fedora"},
-			},
-		}
-
-		withPassword := func(secretName string) libvmi.Option {
-			return func(vmi *v1.VirtualMachineInstance) {
-				vmi.Spec.AccessCredentials = append(vmi.Spec.AccessCredentials,
-					v1.AccessCredential{
-						UserPassword: &v1.UserPasswordAccessCredential{
-							Source: v1.UserPasswordAccessCredentialSource{
-								Secret: &v1.AccessCredentialSecretSource{
-									SecretName: secretName,
-								},
-							},
-							PropagationMethod: v1.UserPasswordAccessCredentialPropagationMethod{
-								QemuGuestAgent: &v1.QemuGuestAgentUserPasswordAccessCredentialPropagation{},
-							},
-						},
-					})
-			}
-		}
+	keysSecretData := libsecret.DataBytes{
+		"my-key1": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"),
+		"my-key2": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"),
+		"my-key3": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"),
+	}
 
-		It("[test_id:6220]should propagate public ssh keys", func() {
-			const secretID = "my-pub-key"
-			vmi := libvmifact.NewFedora(withSSHPK(secretID, withQuestAgentPropagationMethod))
+	userData := fmt.Sprintf(
+		"#cloud-config\npassword: %s\nchpasswd: { expire: False }\n",
+		fedoraPassword,
+	)
 
-			By("Creating a secret with three ssh keys")
-			createNewSecret(testsuite.GetTestNamespace(vmi), secretID, map[string][]byte{
-				"my-key1": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"),
-				"my-key2": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"),
-				"my-key3": []byte("ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"),
-			})
+	DescribeTable("should have ssh-key under authorized keys added", func(secretID string, secretData libsecret.DataBytes, withAccessCredential bool, options ...libvmi.Option) {
```

Comment:
All test cases use the same value for `secretID` and `secretData`. Can you drop the extra table args and use the constants directly instead?